apps/hs/helen/src/Helen/Core.hs

```diff
@@ -79,9 +89,38 @@ initHelen = do
         STM.takeTMVar clientStore >>=
           STM.putTMVar clientStore . HashMap.update (\(c, u) -> Just (c, u >> action cId)) cId
     , _serviceManager = defServiceManager
-    , trustedClients = Set.empty
+    , trustedClients = confTrustedClients conf
     }
 
+readConfigFromYamlFile :: IO Config
+readConfigFromYamlFile = do
+  cf <- resolveFile' "helen-config.yaml"
+  errOrConfig <- readYamlSafe cf
+  case errOrConfig of
+    Left err -> do
+      putStrLn $ unlines
+        [ "WARNING: Failed to read config file: "
+        , err
+        , "pretending that this didn't fail and contained the default config."
+        ]
+      pure mempty
+    Right conf -> pure conf
+
+readConfigFromEnv :: IO Config
+readConfigFromEnv = do
+  mtce <- lookupEnv "TRUSTED_CLIENTS"
+  tcs <- case mtce of
+    Nothing -> pure Set.empty
+    Just clientStr -> fmap Set.fromList $ forM (words clientStr) $ \cs ->
+      case splitOn ":" cs of
+        [ipstr, portstr] -> do
+          ipAddr <- Network.inet_addr ipstr
+          port <- case readMaybe portstr of
+            Nothing -> die $ "Failed to parse port from " <> portstr
+            Just p -> pure p
+          pure $ Network.SockAddrInet port ipAddr
+        _ -> die $ "Failed to parse ip address from " <> cs
```

Comment:
Please, let's avoid halting program if there is any single chance to continue somehow with the error message sent to log :) Failing to parse ip address is not so big deal :)